ImportError: cannot import name 'LaTeXWriter'

### Problem
Between Sphinx 1.7.2 and 1.7.3, the latex builders were moved into a package. Now some of the writers are not importable.

See also https://github.com/Pylons/pyramid/pull/3261

#### Procedure to reproduce the problem
We use Travis-CI and tox to build docs on Python 3.5. [See Travis-CI log for Pyramid](https://travis-ci.org/Pylons/pyramid/jobs/369993136#L543-L561)

#### Error logs / results
```docs runtests: commands[1] | make -C docs doctest html epub BUILDDIR=/home/travis/build/Pylons/pyramid/.tox/docs SPHINXOPTS=-W -E
make: Entering directory `/home/travis/build/Pylons/pyramid/docs'
sphinx-build -b doctest -d /home/travis/build/Pylons/pyramid/.tox/docs/doctrees  -W -E . /home/travis/build/Pylons/pyramid/.tox/docs/doctest
Running Sphinx v1.7.3
Configuration error:
There is a programable error in your configuration file:
Traceback (most recent call last):
  File "/home/travis/build/Pylons/pyramid/.tox/docs/lib/python3.5/site-packages/sphinx/config.py", line 161, in __init__
    execfile_(filename, config)
  File "/home/travis/build/Pylons/pyramid/.tox/docs/lib/python3.5/site-packages/sphinx/util/pycompat.py", line 150, in execfile_
    exec_(code, _globals)
  File "conf.py", line 27, in <module>
    from sphinx.writers.latex import LaTeXTranslator
  File "/home/travis/build/Pylons/pyramid/.tox/docs/lib/python3.5/site-packages/sphinx/writers/latex.py", line 26, in <module>
    from sphinx.builders.latex.transforms import URI_SCHEMES, ShowUrlsTransform  # NOQA  # for compatibility
  File "/home/travis/build/Pylons/pyramid/.tox/docs/lib/python3.5/site-packages/sphinx/builders/latex/__init__.py", line 35, in <module>
    from sphinx.writers.latex import LaTeXWriter, LaTeXTranslator
ImportError: cannot import name 'LaTeXWriter'
make: *** [doctest] Error 2
make: Leaving directory `/home/travis/build/Pylons/pyramid/docs'
ERROR: InvocationError for command '/usr/bin/make -C docs doctest html epub BUILDDIR=/home/travis/build/Pylons/pyramid/.tox/docs SPHINXOPTS=-W -E' (exited with code 2)
```

- https://travis-ci.org/Pylons/pyramid/jobs/369993136#L543-L561

#### Expected results
Docs should pass the build in [Travis-CI like in Sphinx 1.7.2](https://travis-ci.org/Pylons/pyramid/jobs/365457209#L544)

### Reproducible project / your project
- https://github.com/pylons/pyramid

### Environment info
- macOS and Ubuntu 14.04.5 LTS
- Python version: 3.5
- Sphinx version: 1.7.3 (worked fine in 1.7.2)
- tox

--- a/CHANGES
+++ b/CHANGES
@@ -17,6 +17,7 @@ Bugs fixed
 ----------
 
 * #4885, #4887: py domain: Crashed with duplicated objects
+* #4889: latex: sphinx.writers.latex causes recusrive import
 
 Testing
 --------

--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -23,7 +23,6 @@ from six import itervalues, text_type
 
 from sphinx import addnodes
 from sphinx import highlighting
-from sphinx.builders.latex.transforms import URI_SCHEMES, ShowUrlsTransform  # NOQA  # for compatibility
 from sphinx.errors import SphinxError
 from sphinx.locale import admonitionlabels, _
 from sphinx.util import split_into, logging
@@ -2561,3 +2560,10 @@ class LaTeXTranslator(nodes.NodeVisitor):
     def unknown_visit(self, node):
         # type: (nodes.Node) -> None
         raise NotImplementedError('Unknown node: ' + node.__class__.__name__)
+
+
+# Import old modules here for compatibility
+# They should be imported after `LaTeXTranslator` to avoid recursive import.
+#
+# refs: https://github.com/sphinx-doc/sphinx/issues/4889
+from sphinx.builders.latex.transforms import URI_SCHEMES, ShowUrlsTransform  # NOQA

